src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -128,6 +220,57 @@ struct bdf
   }
 };
 
+struct sdm_sensor_info
+{
+  using result_type = query::sdm_sensor_info::result_type;
+  using data_type = query::sdm_sensor_info::data_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type, const boost::any& reqType)
+  {
+    const query::sdr_req_type req_type = boost::any_cast<query::sdm_sensor_info::req_type>(reqType);
+    auto pdev = get_pcidev(device);
+    const std::string target_dir = "hwmon";
+    const std::string target_file = "name";
+    const std::string target_platform = "xilinx_vck5000_gen4x8_xdma_base_1";
```

Comment:
We should not have  platform names encoded here.  Can you find some other way?  Like can't this check (whatever the reason is for the conditional code path) be made data driven?  You see the code is not documented to the point where it is clear what is being done.    Why is there a check on targeted platform, it's a mystery!!!